tests/drivers/flash/testcase.yaml

```diff
@@ -14,3 +14,7 @@ tests:
   drivers.flash.default:
     platform_allow: mimxrt1060_evk
     tags: mcux
+  drivers.flash.stm32:
+    filter: dt_compat_enabled("st,stm32-flash-controller") and
+       dt_compat_enabled_with_label("fixed-partitions", "storage_partition")
```

Comment:
Note that all of stm32 devices that have that compatibility, have `flash0` labeled device, instead of this line you may basically, for devices filtered by the compatibility, add universal overlay that would reorganize partitions, or would be adding the required "storage_partition".